scalameta/dialects/shared/src/main/scala/scala/meta/Dialect.scala

```diff
@@ -0,0 +1,427 @@
+    // Deprecated in 2.10, not supported in 2.11 and newer.
+    val allowCaseClassWithoutParameterList: Boolean,
+    // Are extractor varargs specified using colons, i.e. is `case Extractor(xs: _*)` legal or not?
+    val allowColonForExtractorVarargs: Boolean,
+    // Are enums allowed?
+    // They are in Dotty, but not in Scala 2.12 or older.
+    val allowEnums: Boolean,
+    // Are implicit by name parameters supported?
+    // They are in Dotty, but not in Scala 2.12 or older.
+    val allowImplicitByNameParameters: Boolean,
+    // Are implicit functions supported by this dialect?
+    val allowImplicitFunctionTypes: Boolean,
+    // Are `inline` identifiers supported by this dialect?
+    val allowInlineIdents: Boolean,
+    // Are inline vals and defs supported by this dialect?
+    val allowInlineMods: Boolean,
+    // Are literal types allowed, i.e. is `val a : 42 = 42` legal or not?
+    val allowLiteralTypes: Boolean,
+    // Are method types allowed, i.e. is `(x: X): x.T` legal or not?
+    val allowMethodTypes: Boolean,
+    // Are multiline programs allowed?
+    // Some quasiquotes only support single-line snippets.
+    val allowMultilinePrograms: Boolean,
+    // Are `|` (union types) supported by this dialect?
+    val allowOrTypes: Boolean,
+    // Are unquotes ($x) and splices (..$xs, ...$xss) allowed?
+    // If yes, they will be parsed as patterns.
+    val allowPatUnquotes: Boolean,
+    // Are naked underscores allowed after $ in pattern interpolators, i.e. is `case q"$_ + $_" =>` legal or not?
+    val allowSpliceUnderscores: Boolean,
+    // Are unquotes ($x) and splices (..$xs, ...$xss) allowed?
+    // If yes, they will be parsed as terms.
+    val allowTermUnquotes: Boolean,
+    // Are terms on the top level supported by this dialect?
+    // Necessary to support popular script-like DSLs.
+    val allowToplevelTerms: Boolean,
+    // Are trailing commas allowed? SIP-27.
+    val allowTrailingCommas: Boolean,
+    // Are trait allowed to have parameters?
+    // They are in Dotty, but not in Scala 2.12 or older.
+    val allowTraitParameters: Boolean,
+    // Are type lambdas allowed, i.e. is `[T] => (T, T)` legal or not?
+    val allowTypeLambdas: Boolean,
+    // Are view bounds supported by this dialect?
+    // Removed in Dotty.
+    val allowViewBounds: Boolean,
+    // Are `with` intersection types supported by this dialect?
+    val allowWithTypes: Boolean,
+    // Are XML literals supported by this dialect?
+    // We plan to deprecate XML literal syntax, and some dialects
+    // might go ahead and drop support completely.
+    val allowXmlLiterals: Boolean,
+    // What kind of separator is necessary to split top-level statements?
+    // Normally none is required, but scripts may have their own rules.
+    val toplevelSeparator: String,
+    // Are numeric literal underscore separators, i.e. `1_000_000` legal or not?
+    val allowNumericLiteralUnderscoreSeparators: Boolean
```

Comment:
why not use `= false` here and remove the other ctor?

Reply:
i am guessing, because for binary compatibility, this ctor is private.

Reply:
Default parameter values auto-generate synthetic `$default..` methods on the companion object that leak into the public API. It's easiest to reason about binary compatibility when you type out all the methods by hand, for better or worse.